File: chazm-model/src/main/java/runtimemodels/chazm/model/event/NeedsEventFactory.java
package runtimemodels.chazm.model.event;

import runtimemodels.chazm.api.relation.Needs;

/**
 * The {@linkplain NeedsEventFactory} interface defines the API for constructing {@linkplain NeedsEvent}s.
 *
 * @author Christopher Zhong
 * @since 7.0.0
 */
@FunctionalInterface
public interface NeedsEventFactory {

    /**
     * Constructs an {@linkplain NeedsEvent}.
     *
     * @param category the {@linkplain EventType}.
     * @param needs    the {@linkplain Needs}.
     * @return a {@linkplain NeedsEvent}.
     */
    NeedsEvent build(EventType category, Needs needs);

}

File: chazm-model/src/main/kotlin/runtimemodels/chazm/model/event/NeedsEventFactory.kt
package runtimemodels.chazm.model.event

import runtimemodels.chazm.api.relation.Needs

/**
 * The [NeedsEventFactory] interface defines the API for constructing [NeedsEvent]s.
 *
 * @author Christopher Zhong
 * @since 7.0.0
 */
@FunctionalInterface
interface NeedsEventFactory {

    /**
     * Constructs an [NeedsEvent].
     *
     * @param category the [EventType].
     * @param needs    the [Needs].
     * @return a [NeedsEvent].
     */
    fun build(category: EventType, needs: Needs): NeedsEvent

}
